fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PredicatePropagation.java

```diff
@@ -130,7 +132,7 @@ private Expression doInfer(ComparisonInferInfo equalInfo, ComparisonInferInfo pr
                 .comparisonPredicate.withChildren(newLeft, newRight);
         Expression expr = SimplifyComparisonPredicate.INSTANCE
                 .rewrite(TypeCoercionUtils.processComparisonPredicate(newPredicate), null);
-        return DateFunctionRewrite.INSTANCE.rewrite(expr, null);
+        return DateFunctionRewrite.INSTANCE.rewrite(expr, null).withInferred(true);
     }
```

Comment:
why not put it in construct func with default value

Reply:
yeah, it's a good idea to construct it, i have fixed it